### PR TITLE
feat: halfday and automatic checkout

### PR DESCRIPTION
**Pull request by vijith2121**
updated task.py file

### CODE REVIEW

File: greens/tasks.py [hunk 1]
```diff
@@ -54,3 +68,46 @@ def allocate_leave():
...
+		group_by="log_type",
+		)
+		total_in = 0
+		total_out = 0
+		for row in emp_details:
+			if row.log_type=='IN':
+				total_in = row.count
+			if row.log_type=='OUT':
+				total_out = row.count
+			if str(total_in)>str(total_out):
```
Comment (rtdany10) [inline greens/tasks.py:108]: Dont use string to compare numbers. Use `flt()` instead

File: greens/tasks.py [hunk 2]
```diff
@@ -54,3 +68,46 @@ def allocate_leave():
...
+			if int(total_working_hours) < 5:
+				frappe.throw('Not completed 5 Hours')
+
+	def shift_checkout(doc):
+		emp_details = frappe.get_all('Employee Checkin',filters={
+			'employee': ['=',doc.employee],
+			'time': ['>',today()],
+			'time': ['<', add_to_date(today(), days=1, as_string=True)],
+		},
+		fields=['employee_name', 'employee','count(name) as count','log_type','time'],
```
Comment (rtdany10) [inline greens/tasks.py:98]: What is the purpose of getting `'employee_name', 'employee', 'log_type', 'time'`?

File: greens/tasks.py [hunk 3]
```diff
@@ -54,3 +68,46 @@ def allocate_leave():
...
+		group_by="log_type",
+		)
+		total_in = 0
+		total_out = 0
+		for row in emp_details:
```
Comment (rtdany10) [inline greens/tasks.py:103]: What is the purpose of this variable and this loop? Shouldn't `count` give the required data already?

**Review by rtdany10 — CHANGES_REQUESTED** [review]

File: greens/tasks.py (hunk 1 above)
Comment (vijith2121) [reply greens/tasks.py:108]: changed

File: greens/tasks.py (hunk 2 above)
Comment (vijith2121) [reply greens/tasks.py]: remove unwanted field

File: greens/tasks.py (hunk 3 above)
Comment (vijith2121) [reply greens/tasks.py]: remove loop and update code

**Pull request merged**